### DISCUSSION TITLE
Defined custom element click outside event in Nue?

### DISCUSSION THREAD

**Discussion by x4fingers** [nuejs/nue#242] [Q&A] [answered]
Sorry if I do not understand at all. I'm just testing something on this framework and instantly have this question.

**The first way:**
I'm thinking about defining a custom event like `@click.outside` like Alpine.js was done. Very handy, but I seem to have violated the framework concept while everything follows the web standard. Please me know if any chance this feature will be developed or not.

**The second way:**
I'll implement a function to add an event listener to detect click outside the element:
```
window.clickOutside = (el, callback) => {
  el.addEventListen('click', (event) => {
    // The code...
  }
}
```
each time I need to use it it does not clean in my opinion
```
clickOutside('div', () => {...})
clickOutside('#some-thing', () => {...})
```

So if I want to define a custom event like **element click outside**, how can I do it with vanilla Javascript in Nue to be reusable on both elements and components in the best way?

**Comment by tipiirai** [comment]
Not sure what your goal is but nothing stops you from adding a global event listener to target specific elements. For example:

``` js
document.addEventListener('click', (event) => {
  if (event.target.tagName == 'DIV') {
    // do your thing
  }
})
```

**Reply by x4fingers** [reply]
I know, I just want to ask will you integrate the click outside event handler into Nue.js core in the future (like [this](https://alpinejs.dev/directives/on#outside)), or do I have to do it myself?

**Reply by tipiirai** [reply, marked as answer]
Nue JS is, similar to React, a library for building reactive *components* — so features like those are outside of its context. This kind of click handlers should be implemented with Web Components and/or Vanilla JS. There will be examples once the design system parts are finished and Nue moves forward to reactivity and single-page applications.

**Reply by x4fingers** [reply]
Thank you, I understood.
